### DISCUSSION TITLE
Explicit Method for Large Deformation Model

### DISCUSSION THREAD

**Discussion by FanZhangchen** [idaholab/moose#27359] [Q&A General]
Hi,

Since I would like to build a dislocation density coupled large deformation model, I wonder if it is possible to implement the explicit method to the solid mechanics model. Or is it possible to set the implicit = true while the time integrator is explicit so that I could have a time-explicit only model?

Also, I do not quite understand the linear solver and nonlinear solver in Moose. I am curious about what will the nonlinear solver do in an explicit method.

I do not know too much in this field, if my question is not clear, please let me know.

Best and Thanks,

zc

**Comment by ngrilli** [comment]
@FanZhangchen we have done some work on dislocation dynamics coupled with mechanics, the repository is here:
https://github.com/ngrilli/c_pfor_am

and you can check our recent paper:
https://www.sciencedirect.com/science/article/pii/S002250962300039X

We have used explicit option in the kernels for dislocation transport,
however, I do not recommend to use explicit for the stress equilibrium equation.
You can keep the implicit formulation for the stress equilibrium, while using explicit to update other variables,
such as dislocation density.

Best Regards,
Nicolò

**Reply by GiudGiud** [reply]
>  I am curious about what will the nonlinear solver do in an explicit method.

not much to do if  all the terms are explicit. But you still have an equation albeit fairly easy to solve.
There has been some work on simplifying the solve for fully explicit cases (see ActuallyExplicitEuler and the ExplicitTimeIntegrator time integrator code)

**Reply by FanZhangchen** [reply]
@ngrilli Thanks for your advice. That makes sense to me to use the explicit method for the dislocation density and use implicit for the stress equilibrium equation. I have read your paper and it is beneficial for me. We might meet similar numerical oscillations caused by the dislocation density transport equation and I am also trying to solve it with DG.

@GiudGiud Thanks for your reply. I will check the code. And I have another question about the off-diagonal terms in the stiffness matrix of Moose. Maybe I ignored some details, but I didn't find the document that describes how Moose treats the coupled off-diagonal terms. Are there any descriptions of how Moose obtains the coupled off-diagonal terms in the stiffness matrix? I just want to have a further understanding of the coupled terms in Moose. Thank you.
